tracer/src/Datadog.Trace/Configuration/ConfigurationSources/GlobalConfigurationSource.cs

```diff
@@ -44,14 +47,14 @@ internal static CompositeConfigurationSource CreateDefaultConfigurationSource()
         return configurationSource;
     }
 
-    internal static bool TryLoadJsonConfigurationFile(IConfigurationSource configurationSource, string baseDirectory, out IConfigurationSource jsonConfigurationSource)
+    internal static bool TryLoadJsonConfigurationFile(IConfigurationSource configurationSource, string? baseDirectory, [NotNullWhen(true)] out IConfigurationSource? jsonConfigurationSource)
     {
         try
         {
             // if environment variable is not set, look for default file name in the current directory
             var configurationFileName = configurationSource.GetString(ConfigurationKeys.ConfigurationFileName) ??
                                         configurationSource.GetString("DD_DOTNET_TRACER_CONFIG_FILE") ??
-                                        Path.Combine(baseDirectory ?? GetCurrentDirectory(), "datadog.json");
+                                        Path.Combine(baseDirectory ?? GetCurrentDirectory() ?? Directory.GetCurrentDirectory(), "datadog.json");
```

Comment:
NIT: the `?? Directory.GetCurrentDirectory()` part should be in `GetCurrentDirectory()`